### ISSUE TITLE
Some issues with headless.py

### ISSUE THREAD

**Issue by ny4i** [n1kdo/n1mm_view#27]
Jeff - I found some issues with headless and I needed to change not to make it run. I will send a pull request with my changes but I wanted to confirm something.

I could not figure out where the code to setup the directory not he RAM disk went then I realized you changed the INIT to create the directory at start-up. Is that the idea so the script does not have to create the directory? I added that back in as my original changes had it. 

Is it OK to leave that in as a check so if someone does not have the INIT script to create the directory in RAM that the script will create them?

Thanks - Tom

**Comment by n1kdo** [comment]
Hi,  I don't actually run it headless. ke0pci is your man.  He made those changes.

**Comment by ke0pci** [comment]
Tom,

I think something happened with autocorrect when you were typing that.
What did you need to change for headless to work?  Most of my changes were
around getting the base install for Raspberry Pi down to 2-3 commands.  I
am happy to revisit the RAMdisk setup to make sure it works for all cases.
I just need to make sure I see it clearly.

John
----
From the global headquarters of amateur radio station KE0PCI

**Comment by ny4i** [comment]
I re-read my message and other than "on the RAM..." being changed to "not on RAM...", no typo.so I hope that was what you meant.

Bottom line was that headless as checked in did not work. I had to remove base_map in the draw_map call.

I am working on a PULL request and will send up my changes later today.

I also had to change how create_images was called as it was not setting the last_qso_time to the result of the function.

I added code to ensure the directory for the HTML files exists. if not, I create it. If the RAM disk init is running, then that will not be necessary but it is a good validation.

My application is that I need headless for HTML and the dashboard for a local display so I am plan to run both on the same Pi along with the collector. I had these both in dashboard when I first wrote the png generation code, but they were broken out to modularize it. I see the point but double the database reads. 

Thanks,
Tom

**Comment by ke0pci** [comment]
Tom,

The ramdisk path changed based on the preferred instructions.  I apparently
did not read through the headless.py file when I made that update.  I am
looking forward to seeing your changes and will scan all files in my future
branch changes as a way of improving.

73, John
----
From the global headquarters of amateur radio station KE0PCI

**Comment by ny4i** [comment]
No problem.
I only discovered that because I did not realize there was a new INIT to build the RAM disk at the start. We should put a comment in config.py that the volume should be /mnt/ramdisk/ to match up with the Ramdisk init.

When I ran headless, it received errors writing since the directories were not there (I had not run the script).

I think I still prefer the script controlling its own destiny so to speak and creating the directories instead of the INIT script but I understand the purpose.

Should be tonight for the changes as have to take out my customizations for my club and location and some debug calls.

Tom
Principal Solutions Architect
Better Software Solutions, Inc.
727-437-2771
